02-microarray/gene-id-convert_microarray_01_ensembl.Rmd

```diff
@@ -0,0 +1,272 @@
+
+
+# {{NAME OF example}}
+
+Ensembl IDs can be used to obtain various different annotations at the 
+gene/transcript level.
+Although this example script uses Ensembl IDs from Zebrafish, (<i>Danio rerio</i>),
+to obtain gene symbols this script can be easily converted for use with different 
+species or annotation types <i>eg</i> protein IDs, gene ontology, accession 
+numbers. 
+
+<b>For different species</b>, wherever the abbreviation `org.Dr.eg.db` or `Dr` is 
+written, it must be replaced with the respective species abbreviation <i>eg</i>
+for<i> Homo sapiens</i> `org.Hs.eg.db` or `Hs` would be used.
+A full list of the annotation R packages from Bioconductor is at this [link](https://bioconductor.org/packages/release/bioc/vignettes/limma/inst/doc/usersguide.pdf).
+
+## Install libraries
+
+See our Getting Started page with [instructions for package installation](https://alexslemonade.github.io/refinebio-examples/01-getting-started/getting-started.html#what-you-need-to-install) for a list of the other software you will need, as well as more tips and resources.
+
+In this analysis, we will be using the `org.Mm.eg.db` R package [@Carlson2020].
+
+```{r}
+# Install the mouse package
+if (!("org.Mm.eg.db" %in% installed.packages())) {
+  # Install this package if it isn't installed yet
+  BiocManager::install("org.Mm.eg.db", update = FALSE)
+}
+```
+
+Attach the packages we need for this analysis.
+
+```{r}
+# Attach the library
+library(org.Mm.eg.db)
+
+# We will need this so we can use the pipe: %>%
+library(magrittr)
+```
+
+## Import and set up data
+
+Data downloaded from refine.bio include a metadata tab separated values (TSV) file and a data TSV file. 
+This chunk of code will read the both TSV files and add them as data frames to your environment.
+
+```{r}
+# Read in metadata TSV file
+metadata <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your metadata file
+  "metadata_GSE13490.tsv" # Replace with the name of your metadata file
+))
+
+# Read in data TSV file
+df <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your data file
+  "GSE13490.tsv" # Replace with the name of your data file
+)) %>%
+  # Tuck away the Gene id column as rownames
+  tibble::column_to_rownames("Gene")
+```
+
+Let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>%
+  dplyr::select(metadata$geo_accession)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$geo_accession)
+
+# Bring back the "Gene" column in preparation for mapping
+df <- df %>%
+  tibble::rownames_to_column("Gene")
+```
+
+
+## Map Ensembl IDs to gene symbols
+
+This code chunk will return a message that indicates that there are many mappings between our keys and columns but this is okay and is what we expect since genes and transcripts do not have a 1:1 relationship.
```

Comment:
I think we will need to hit this point harder and briefly mention the different options for `multiVals` argument, highlighting that different downstream analyses may benefit from particular strategies. Whatever strategy we end up using here, we should also use a different one when we update the rna-seq conversion module.